src/responder/kcm/kcmsrv_ccache_secdb.c

```diff
@@ -818,10 +822,176 @@ static errno_t ccdb_secdb_get_default_recv(struct tevent_req *req,
     return EOK;
 }
 
+static errno_t ccdb_secdb_get_cc_for_uuid(size_t uuid_list_count,
+                                          const char **uuid_list,
+                                          const char **uid_list,
+                                          struct ccdb_secdb *secdb,
+                                          struct kcm_ccache **cc_list,
+                                          int *_new_count)
+{
+    TALLOC_CTX *tmp_ctx;
+    errno_t ret;
+    uid_t uid;
+    char **list;
+    uuid_t uuid;
+    char *uuid_str;
+    char *secdb_key;
+    struct cli_creds cli_cred;
+    int real_count = 0;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        ret = ENOMEM;
+        return ret;
+    }
+
+    for (size_t i = 0; i < uuid_list_count; i++) {
+        struct passwd *pwd;
+
+        cc_list[i] = talloc_zero(cc_list, struct kcm_ccache);
+        if (cc_list[i] == NULL) {
+            ret = ENOMEM;
+            goto done;
+        }
+
+        ret = split_on_separator(tmp_ctx, uuid_list[i], ':', true, true,
+                                 &list, NULL);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE, "split on separator failed [%d]: %s\n",
+                                     ret, sss_strerror(ret));
+            goto done;
+        }
+
+        uuid_str = list[0];
+        uuid_str[UUID_STR_SIZE - 1] = '\0';
+        ret = uuid_parse(uuid_str, uuid);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE, "uuid parse of [%s] failed [%d]: %s\n",
+                                     list[0], ret, sss_strerror(ret));
+            goto done;
+        }
+        uid = strtouint32(uid_list[i], NULL, 10);
+        ret = errno;
+        if (ret != EOK) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Invalid UID [%s] conversion to uint32 "
+                                       "[%d]: %s\n", uid_list[i], ret,
+                                       sss_strerror(ret));
+            goto done;
+        }
+
+        errno = 0;
+        pwd = getpwuid(uid);
+        if (pwd == NULL) {
+            ret = errno;
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to get pwd entry for "
```

Comment:
SSSDBG_MINOR_FAILURE since it is not impotant. It will not also invoke the new backtrace functionality.

Reply:
Perhaps also change the message to something more useful for users like "Unable to resolve user [uid] who is the owner of an existing ccache [%d]: %s\n"

Reply:
Fixed.